likeHack.js

```diff
@@ -0,0 +1,111 @@
+import mongoose from "mongoose";
+
+const uri = `mongodb://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@ds157136.mlab.com:57136/hackone`;
+let conn = null;
+
+export const like = async (event, context) => {
+  //need a hackid
+  //need a userid
+  //keep an array of likes in hack collecton's document
+  //when user likes put in the likes array
+  //when user unlike, take user out of the likes array
+  context.callbackWaitsForEmptyEventLoop = false;
+  const data = JSON.parse(event.body);
+  const { hackId, userId } = data;
+
+  if (conn == null) {
+    conn = await mongoose.createConnection(uri, {
+      bufferCommands: false,
+      bufferMaxEntries: 0
+    });
+    conn.model(
+      "Hack",
+      new mongoose.Schema({
+        likes: { type: [mongoose.ObjectId], default: [] },
+        numberLikes: { default: 0, type: Number }
```

Comment:
this seems easy enough to derive in JS code, and easy to get out of sync... I would not be storing this in the DB.

Reply:
got it, will try to return the length of the likes array! and just either add or subtract from that if user dislike/like the hack